src/relocatable.rs

```diff
@@ -191,13 +192,21 @@ impl From<PyRelocatable> for PyMaybeRelocatable {
 
 impl From<&BigUint> for PyMaybeRelocatable {
     fn from(val: &BigUint) -> Self {
-        PyMaybeRelocatable::Int(val.clone())
+        PyMaybeRelocatable::Int(if val < &CAIRO_PRIME {
+            val.clone()
+        } else {
+            val % &*CAIRO_PRIME
+        })
```

Comment:
This looks like `PyMaybeRelocatable` should simply contain a `Felt` and let it handle the conversion.

Reply:
We'd have to make `Felt` a pyclass for that to work